core/src/main/java/org/ehcache/events/CacheEventNotificationServiceImpl.java

```diff
@@ -122,112 +174,162 @@ public void setStoreListenerSource(Cache<K, V> source) {
   }
 
   @Override
-  public void onEvent(final CacheEvent<K, V> event) {
+  public void onEvent(CacheEvent<K, V> event) {
```

Comment:
We need to find a way to bypass completely the creation of the thread locals when there are no registered listeners.